src/main/java/com/internet/shop/dao/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,190 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.exceptions.DataBaseProcessingException;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.models.Product;
+import com.internet.shop.models.ShoppingCart;
+import com.internet.shop.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ShoppingCartDaoJdbcImpl implements ShoppingCartDao {
+    @Override
+    public Optional<ShoppingCart> getByUserId(Long userId) {
+        ShoppingCart shoppingCart = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String query = "SELECT id_shoping_cart "
+                    + "FROM shoping_cart "
+                    + " WHERE id_user = ? AND deleted = FALSE";
```

Comment:
```suggestion
            String query = "SELECT id_shoping_cart FROM shoping_cart "
                    + " WHERE id_user = ? AND deleted = FALSE";
```

Reply:
fixed query in  getByUserId() method in ShoppingCartDaoJdbcImpl